README.md

```diff
@@ -1,2 +1,208 @@
+`pre-build`, `standard-build`, and `post-build`.  The common recipes only define
+actions for `standard-` sub-target.  The others are provided to allow custom actions.
+
+### Variables
+
+These variables are used by the standard targets and may be overridden:
+
+* __PROJECT__ - The name of the project being built.  Defaults to the name of the current directory (which should be also the name of the repository)
+* __BUILDDIR__ - The location that build artifacts will be placed in.  Defaults to `./build`
+* __GOTARGETS__ - A list of the executables to build.  Defaults to the project name
+* __GOCMDDIR__ - The location of the directory containing the sub-directories for the executables to be built.  Defaults to `./cmd`
+* __GOROOTTARGET__ - Can be set to the name of a single target from `GOTARGETS` if the main executable code for it is in the top level directory.  No default
+* __GOSRC__ - A list of the Go source files.  Defaults to all Go source files recursively found in the current directory
+* __GO__ - The Go executable to run.  Defaults to `go`
+* __GOFLAGS__ - Flags to pass to `go build`.  No default
+* __GOTESTTARGET__ - The target to pass to `go test`.  Defaults to `./...`, which means all tests in the project
+* __GOTESTFLAGS__ - Flags to pass to `go test`.  Defaults to `-v -race`
+
+
+## Copy
+
+Recipes for copying files as part of the build are included in `copy-common.mk`
+
+### Targets
+
+The primary targets provided for Go are:
+
+* __build__ - Copies files into the build directory
+* __clean__ - Removes build targets
+
+Each primary target depends on three sub-targets, which use the prefixes `pre-`,
+`standard-`, and `post-`.  E.g., the sub-targets for the `build` target are
+`pre-build`, `standard-build`, and `post-build`.  The common recipes only define
+actions for `standard-` sub-target.  The others are provided to allow custom actions.
+
+### Variables
+
+These variables are used by the standard targets and may be overridden:
+
+* __PROJECT__ - The name of the project being built.  Defaults to the name of the current directory (which should be also the name of the repository)
+* __BUILDDIR__ - The location that build artifacts will be placed in.  Defaults to `./build`
+* __COPY__ - The program to use to copy files.  Defaults to `cp`
+* __COPYTARGETS__ - A list of files and directories to be copied into `BUILDDIR`.  No default
+* __COPYFLAGS__ - Flags to pass to `COPY`.  Defaults to `-r`
+
+## Using
+
+The normal way to use the common make recipes is to include the appropriate files in
+the Makefile for a repository.  For a simple Go microservice, just including
+`go-common.mk` may be all that is required in the Makefile, for example:
+
+```make
+include go-common.mk
+```
+
+### Include
+
+In all cases the Makefile will ultimately reference one or more of the `*-common.mk`
+files in this package, usually through the use of the `include` directive.  However,
+there are a number of options on how this directive can be resolved.
```

Comment:
I don't know if choosing between the different inclusion options is part of the point of this change or if it is meant to remain open to users but for me personally I favor having a local copy of it to ease offline dev and discovery.

A medium solution could also be downloading it locally but have it check if it is out of date and print that in the console as a warning.

Reply:
> I don't know if choosing between the different inclusion options is part of the point of this change or if it is meant to remain open to users but for me personally I favor having a local copy of it to ease offline dev and discovery.
> 
> A medium solution could also be downloading it locally but have it check if it is out of date and print that in the console as a warning.

I think in general that I agree that having a local copy makes better sense, particularly if we also have a way to detect that we're out of date.  I'll update the guidance and see if such a target is simple to create.